### PR TITLE
add input devices

### PR DESCRIPTION
**Pull request by JanJaapKo**
@dylian94 ,

I've updated quite some things on your plugin:
- I separated all the GoodWe stuff into a separate  module with some new classes which I think reflect more the design of the API
- I added Domoticz devices for the various input strings (I happen to have 2 strings, 1 east and 1 west, works very nice)
- I created a status switch for the status of the inverter

I hope you can approve this PR. Or let me know if something needs improving.

Thanks, Jan-Jaap

### CODE REVIEW

**Review by djoenez — APPROVED** [review]

File: plugin.py [hunk 1]
```diff
@@ -188,64 +164,98 @@ def onMessage(self, Connection, Data):
...
 
-                Connection.Send(self.stationDataRequest())
+                Connection.Send(self.goodWeInverter.stationDataRequest())
 
             elif "api/v2/PowerStation/GetMonitorDetailByPowerstationId" in apiUrl:
+                Domoticz.Debug("message received: GetMonitorDetailByPowerstationId")
                 if apiData is None:
-                    Domoticz.Error("No station data received from GoodWe SEMS API (Station ID: " + self.powerStationList[
+                    Domoticz.Error("No station data received from GoodWe SEMS API (Station ID: " + self.goodWeInverter.powerStationList[
                         self.powerStationIndex] + ")")
```
Comment (djoenez) [inline plugin.py:194]: Missing: self.GoodWeInverter.powerStationIndex

**Comment by djoenez** [comment]
I tried your addition, i only get error:
Async Secure Read Exception: 335544539, short read 

in the logs, i havent dabbled much in JSON, do you know where this comes from?

**Comment by JanJaapKo** [comment]
@djoenez  I've seen the same error, doesn't seem to hurt. Several mentions of this error on the Domoticz forum, no-one seems to have a solution. Oddly, it goes away when the plugin runs in Debug mode. Will see if I can do something about it in the disconnect.

**Comment by JanJaapKo** [comment]
@djoenez , @dylian94 I created #9 to add some exception handling. My Git skills aren't good enough to add them to this PR, so I will close it.

**Pull request closed without merging**